src/course-outline/card-header/CardHeader.jsx

```diff
@@ -91,134 +97,148 @@ const CardHeader = ({
+                  {intl.formatMessage(messages.menuCopy)}
+                </Dropdown.Item>
+              )}
+              {actions.duplicable && (
                 <Dropdown.Item
-                  data-testid={`${namePrefix}-card-header__menu-move-up-button`}
-                  onClick={onClickMoveUp}
-                  disabled={!actions.allowMoveUp}
+                  data-testid={`${namePrefix}-card-header__menu-duplicate-button`}
+                  onClick={onClickDuplicate}
                 >
-                  {intl.formatMessage(messages.menuMoveUp)}
+                  {intl.formatMessage(messages.menuDuplicate)}
                 </Dropdown.Item>
+              )}
+              {actions.draggable && (
+                <>
+                  <Dropdown.Item
+                    data-testid={`${namePrefix}-card-header__menu-move-up-button`}
+                    onClick={onClickMoveUp}
+                    disabled={!actions.allowMoveUp}
+                  >
+                    {intl.formatMessage(messages.menuMoveUp)}
+                  </Dropdown.Item>
+                  <Dropdown.Item
+                    data-testid={`${namePrefix}-card-header__menu-move-down-button`}
+                    onClick={onClickMoveDown}
+                    disabled={!actions.allowMoveDown}
+                  >
+                    {intl.formatMessage(messages.menuMoveDown)}
+                  </Dropdown.Item>
+                </>
+              )}
+              {actions.deletable && (
                 <Dropdown.Item
-                  data-testid={`${namePrefix}-card-header__menu-move-down-button`}
-                  onClick={onClickMoveDown}
-                  disabled={!actions.allowMoveDown}
+                  className="border-top border-light"
+                  data-testid={`${namePrefix}-card-header__menu-delete-button`}
+                  onClick={onClickDelete}
                 >
-                  {intl.formatMessage(messages.menuMoveDown)}
+                  {intl.formatMessage(messages.menuDelete)}
                 </Dropdown.Item>
-              </>
-            )}
-            {actions.deletable && (
-              <Dropdown.Item
-                className="border-top border-light"
-                data-testid={`${namePrefix}-card-header__menu-delete-button`}
-                onClick={onClickDelete}
-              >
-                {intl.formatMessage(messages.menuDelete)}
-              </Dropdown.Item>
-            )}
-          </Dropdown.Menu>
-        </Dropdown>
+              )}
+            </Dropdown.Menu>
+          </Dropdown>
+        </div>
       </div>
-    </div>
+      <Sheet
+        position="right"
+        show={isManageTagsDrawerOpen}
+        onClose={/* istanbul ignore next */ () => closeManageTagsDrawer()}
+      >
+        <ContentTagsDrawer
+          id={cardId}
+          onClose={/* istanbul ignore next */ () => closeManageTagsDrawer()}
+        />
+      </Sheet>
+    </>
   );
```

Comment:
I moved the Drawer, the tagCount and the related states and events to the `CardHeader`.

Let me know what you think!